twake/frontend/src/app/scenes/Apps/Drive/Viewer/Viewer.tsx

```diff
@@ -111,7 +115,25 @@ export default class Viewer extends Component<PropsType, StateType> {
             {!DriveService.previewonly && editor_candidate.length === 1 && (
               <Button
                 className="small open-with"
-                onClick={() => this.openFile(editor_candidate[0])}
+                onClick={() => {
+                  if (FeatureTogglesService.isActiveFeatureName(FeatureNames.ONLY_OFFICE)) {
+                    this.openFile(editor_candidate[0]);
+                  } else {
+                    ModalManager.open(
+                      <LockedOnlyOfficePopup
+                        pricingPlanUrl={
```

Comment:
Can't we put this inside the LockedOnlyOfficePopup instead of having a props ? @stephanevieira75

Reply:
Done